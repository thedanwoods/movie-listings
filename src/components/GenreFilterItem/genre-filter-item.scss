.genre-filter-item {
  list-style-type: none;
  padding: 2px;
  margin: 0;
}

.genre-filter-item__checkbox input[type='checkbox'] {
  opacity: 0;
  cursor: pointer;
  position: absolute;
}

.genre-filter-item__text {
  background-color: #2a1a1f;
  padding: 4px;
  box-sizing: content-box;
  text-transform: uppercase;
  font-family: 'Montserrat', sans-serif;
  font-weight: 400;
  font-size: 11px;
  cursor: pointer;
  &:hover {
    background-color: #3a242b
  }
}

input[type=checkbox]:checked ~ .genre-filter-item__text {
  background: #ad8350;
}

input[type=checkbox]:focus ~ .genre-filter-item__text, 
input[type=checkbox]:active ~ .genre-filter-item__text {
  padding: 3px;
  border: .5px dotted #fff;
}
