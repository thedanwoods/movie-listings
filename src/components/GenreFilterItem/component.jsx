import React from 'react';
import PropTypes from 'prop-types';

import './genre-filter-item.scss';

const GenreFilterItem = ({ genre, onClick, checked }) => (
  <li className="genre-filter-item">
    <label
      className="genre-filter-item__checkbox"
      htmlFor={`genre-filter-${genre.name}`}
    >
      <input
        type="checkbox"
        id={`genre-filter-${genre.name}`}
        name={genre.name}
        onChange={onClick}
        checked={checked}
      />
      <div className="genre-filter-item__text">{genre.name}</div>
    </label>
  </li>
);

GenreFilterItem.propTypes = {
  genre: PropTypes.object.isRequired,
  checked: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default GenreFilterItem;
