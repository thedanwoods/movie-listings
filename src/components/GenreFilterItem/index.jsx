import { connect } from 'react-redux';
import GenreFilterItem from './component';
import checkboxClicked from '../../actions/checkboxClicked';

export const mapStateToProps = (state, ownProps) => ({
  checked: state.genreFilter.includes(ownProps.genre.id),
});

export const mapDispatchToProps = (dispatch, ownProps) => ({
  onClick: () => {
    dispatch(checkboxClicked(ownProps.genre.id));
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(GenreFilterItem);
