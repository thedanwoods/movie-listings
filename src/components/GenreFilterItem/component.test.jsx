import React from 'react';
import { shallow } from 'enzyme';
import GenreFilterItem from './component';

const defaultProps = {
  genre: { id: 1, name: 'Action' },
  onClick: jest.fn(),
  checked: false,
};

describe('<GenreFilterItem /> component', () => {
  it('renders without errors', () => {
    const genreFilterItem = shallow(<GenreFilterItem {...defaultProps} />);

    expect(() => genreFilterItem).not.toThrow();
    expect(genreFilterItem.debug()).toMatchSnapshot();
  });
});
