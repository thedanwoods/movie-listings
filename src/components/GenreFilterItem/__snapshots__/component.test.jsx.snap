// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenreFilterItem /> component renders without errors 1`] = `
"<li className=\\"genre-filter-item\\">
  <label className=\\"genre-filter-item__checkbox\\" htmlFor=\\"genre-filter-Action\\">
    <input type=\\"checkbox\\" id=\\"genre-filter-Action\\" name=\\"Action\\" onChange={[Function: mockConstructor]} checked={false} />
    <div className=\\"genre-filter-item__text\\">
      Action
    </div>
  </label>
</li>"
`;
